runtime/doc/lua.txt

```diff
@@ -1776,6 +1743,44 @@ vim.schedule_wrap({fn})                                  *vim.schedule_wrap()*
       • |vim.schedule()|
       • |vim.in_fast_event()|
 
+                                                         *vim.str_byteindex()*
+vim.str_byteindex({s}, {encoding}, {index}, {no_out_of_range})
+    Convert UTF-32, UTF-16 or UTF-8 {index} to byte index. If
+    {no_out_of_range} is true then then an out of range index will return byte
+    length instead of throwing an error.
+
+    Invalid UTF-8 and NUL is treated like in |vim.str_utfindex()|. An {index}
+    in the middle of a UTF-16 sequence is rounded upwards to the end of that
+    sequence.
+
+    Parameters: ~
+      • {s}                (`string`)
+      • {encoding}         (`"utf-8"|"utf-16"|"utf-32"`)
+      • {index}            (`integer`)
+      • {no_out_of_range}  (`boolean?`)
+
+    Return: ~
+        (`integer`)
+
+                                                          *vim.str_utfindex()*
+vim.str_utfindex({s}, {encoding}, {index}, {no_out_of_range})
```

Comment:
the old signature started with `s, index`. was that changed intentionally? would it be helpful for migrating to avoid changing the order where possible? (if this was already discussed/resolved, LMK).

```suggestion
vim.str_utfindex({s}, {index}, {encoding}, {no_out_of_range})
```

Reply:
> We can support the old versions by dispatching on > the type of argument 2, and deprecating that form

Discussed here
https://github.com/neovim/neovim/issues/30804#issuecomment-2411675761